src/json.h

```diff
@@ -156,7 +156,7 @@ namespace abc {
 
 		if (size < sizeof(json::token_t)) {
 			char buffer[100];
-			std::snprintf(buffer, sizeof(buffer), "json_istream::get_token() size=%ld (< %ld) ", (std::int32_t)size, (std::int32_t)sizeof(json::token_t));
+			std::snprintf(buffer, sizeof(buffer), "json_istream::get_token() size=%zd (< %zd) ", size, sizeof(json::token_t));
```

Comment:
Please make both of these "%zu" as `size_t` is unsigned by definition.

Reply:
As discussed, here are the build warnings I was seeing. This on Native x64 Linux (Ubuntu) and on x64 Windows 10 with Ubuntu WSL;

# ---------- Begin building basic ----------
mkdir /home/nash/Projects/samples/abc-nesho-clone/abc/out/samples/basic
g++ -ggdb --std=c++17 -Wpedantic -o /home/nash/Projects/samples/abc-nesho-clone/abc/out/samples/basic/basic /home/nash/Projects/samples/abc-nesho-clone/abc/samples/basic/*.cpp -l:libstdc++.so.6 -l:libgcc_s.so.1 -l:libpthread.so
In file included from /home/nash/Projects/samples/abc-nesho-clone/abc/samples/basic/equations.h:29,
                 from /home/nash/Projects/samples/abc-nesho-clone/abc/samples/basic/main.cpp:32:
/home/nash/Projects/samples/abc-nesho-clone/abc/samples/basic/../../src/json.h: In instantiation of ‘void abc::json_istream<MaxLevels, Log>::get_token(abc::json::token_t*, std::size_t) [with long unsigned int MaxLevels = 64; Log = abc::log_ostream<abc::debug_line_ostream<>, abc::log_filter>; std::size_t = long unsigned int]’:
/home/nash/Projects/samples/abc-nesho-clone/abc/samples/basic/equations.h:133:4:   required from ‘void abc::samples::equations_endpoint<Limits, Log>::process_rest_request(abc::http_server_stream<Log>&, const char*, const char*) [with Limits = abc::endpoint_limits; Log = abc::log_ostream<abc::debug_line_ostream<>, abc::log_filter>]’
/home/nash/Projects/samples/abc-nesho-clone/abc/samples/basic/equations.h:59:14:   required from here
/home/nash/Projects/samples/abc-nesho-clone/abc/samples/basic/../../src/json.h:159:76: warning: format ‘%ld’ expects argument of type ‘long int’, but argument 4 has type ‘int32_t’ {aka ‘int’} [-Wformat=]
  159 |    std::snprintf(buffer, sizeof(buffer), "json_istream::get_token() size=%ld (< %ld) ", (std::int32_t)size, (std::int32_t)sizeof(json::token_t));
      |                                                                          ~~^
      |                                                                            |
      |                                                                            long int
      |                                                                          %d
/home/nash/Projects/samples/abc-nesho-clone/abc/samples/basic/../../src/json.h:159:83: warning: format ‘%ld’ expects argument of type ‘long int’, but argument 5 has type ‘int32_t’ {aka ‘int’} [-Wformat=]
  159 |    std::snprintf(buffer, sizeof(buffer), "json_istream::get_token() size=%ld (< %ld) ", (std::int32_t)size, (std::int32_t)sizeof(json::token_t));
      |                                                                                 ~~^
      |                                                                                   |
      |                                                                                   long int
      |                                                                                 %d
mkdir /home/nash/Projects/samples/abc-nesho-clone/abc/out/samples/basic/resources
cp /home/nash/Projects/samples/abc-nesho-clone/abc/samples/basic/resources/* /home/nash/Projects/samples/abc-nesho-clone/abc/out/samples/basic/resources
# ---------- Done building basic ----------